File: python/program/Ram.py
from Komponen import Komponen
from Memory import Memory


class Ram(Komponen, Memory):
    __ddr = ""

    def __init__(self, ddr, kapasitasGB, kecepatanMHz, merk, nama):
        Komponen.__init__(self, merk, nama)
        Memory.__init__(self, kapasitasGB, kecepatanMHz)
        self.__ddr = ddr

    def set_ddr(self, ddr):
        self.__ddr = ddr

    def get_ddr(self):
        return self.__ddr

File: cpp/program/Ram.cpp
#pragma once
#include "Komponen.cpp"
#include "Memory.cpp"
#include <iostream>
#include <string>

using namespace std;

class Ram : public Komponen, public Memory {
  private:
    string ddr;

  public:
    Ram() {}

    Ram(string ddr, int kecepatanMHz, int kapasitasGB, string merk, string nama)
        : Komponen(merk, nama), Memory(kapasitasGB, kecepatanMHz) {
        this->ddr = ddr;
    }

    void setDdr(string ddr) {
        this->ddr = ddr;
    }

    string getDdr() {
        return this->ddr;
    }

    ~Ram() {
    }
};
